pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -275,40 +276,38 @@ func (p *DNSProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 	requestID := request.Id // keep the original request ID
 	qname := string(request.Question[0].Name)
 	protocol := w.LocalAddr().Network()
-	endpointAddr := w.RemoteAddr().String()
 	scopedLog := log.WithFields(logrus.Fields{
 		logfields.DNSName:      qname,
 		logfields.IPAddr:       w.RemoteAddr(),
 		logfields.DNSRequestID: request.Id})
 
-	scopedLog.Debug("Handling DNS query from endpoint")
```

Comment:
Could you leave this debug line in? It's useful when diagnosing whether the proxy is being slow to begin handling packets or something else is slow in NotifyOnDNSMsg.